plugins/modules/archive.py

```diff
@@ -534,7 +538,7 @@ def _get_checksums(self, path):
             archive = zipfile.ZipFile(_to_native_ascii(path), 'r')
             checksums = set((info.filename, info.CRC) for info in archive.infolist())
             archive.close()
-        except zipfile.BadZipfile:
+        except zipfile.BadZipFile:
```

Comment:
With the `zipfile.` prefix you're not using the one you just imported :)
```suggestion
        except BadZipFile:
```